### PR TITLE
AbstractPlatform: allow any string in interval for date interval expression

### PR DESCRIPTION
**Pull request by ondrejmirtes**
|      Q       |   A
|------------- | -----------
| Type         | improvement

#### Summary

This change allows these methods to be called with just a `string` in ORM 3.0.x. See https://github.com/doctrine/orm/pull/11243#issuecomment-1936232509

DBAL 4.0.x already has `string` there.

Also I'd like to note that I noticed that methods `getDateAddQuartersExpression` and `getDateSubQuartersExpression` aren't called from ORM. Shouldn't support for them be added?

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
There are going to be some CI failures, waiting for the workflow approval :)

**Review by derrabus — DISMISSED** [review]

**Comment by derrabus** [comment]
Static analysis is failing as predicted. 🙂

**Comment by ondrejmirtes** [comment]
Alright, the 2nd commit backports the solution from DBAL 4.0 :) Let's run the workflows again.

**Comment by ondrejmirtes** [comment]
The CS violation should be gone now.

**Comment by ondrejmirtes** [comment]
Alright, it's green :) Once this is merged and released, I'll update the DBAL version constraint https://github.com/doctrine/orm/pull/11243 to require this new version.

**Review by derrabus — APPROVED** [review]

**Pull request merged**
